src/gui/src/Skin.h

```diff
@@ -68,12 +68,41 @@ class Skin
 	 */
 	static QString getWarningButtonStyleSheet( int nSize );
 
+	/**
+	 * Draws the background of a row in both the pattern list of the
+	 * SongEditor and the instrument list in the PatternEditor using
+	 * @a p.
+	 *
+	 * \param p Painter used in the calling QPaintEvent routine.
+	 * \param rect Boundary that encloses element (one row).
+	 * \param background Color used.
+	 * \param bHovered Whether the element is currently hovered by mouse.
+	 */
+	static void drawListBackground( QPainter* p, QRect rect, QColor background,
+									bool bHovered );
 	/** If a widget is marked inactive the value of its background
 		color are reduced by this factor.*/
 	static QColor makeWidgetColorInactive( QColor color );
 		/** If a widget is marked inactive the value of its text color
 		are reduced by this factor.*/
 	static QColor makeTextColorInactive( QColor color );
+
+	static constexpr int nPlayheadWidth = 11;
+	static constexpr int nPlayheadHeight = 8;
+	static constexpr int getPlayheadShaftOffset() {
+		return std::floor( Skin::nPlayheadWidth / 2 ); }
```

Comment:
`std::floor` isn't guaranteed to be `constexpr` [until C++23](https://en.cppreference.com/w/cpp/numeric/math/floor). Seems it's not `constexpr` on macOS.

Reply:
Oh, rats, looks like the Appveyor builds completed successfully despite it failing to build. That's almost certainly my fault, sorry.